wagtail/snippets/action_menu.py

```diff
@@ -298,6 +299,32 @@ def __init__(self, request, **kwargs):
             if menu_item.is_shown(self.context)
         ]
 
+        if instance and isinstance(instance, WorkflowMixin):
```

Comment:
Isn't this pretty much identical to the `Page` logic? Could it be shared?

Reply:
Yes, but at this point I think it's too risky to unify them together. We have an existing issue to redesign this (#9030, #9869), which I think would be an excellent time to unify the two.